.github/workflows/ci.yaml

```
@@ -121,3 +121,33 @@ jobs:
         run: |
           . .venv/bin/activate
           make test
+
+  main-java:
+    name: CI [Java]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: ["11", "17", "21"]
+    steps:
+      - name: Install SSH Key
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          known_hosts: ${{ vars.SSH_KNOWN_HOSTS }}
+      - name: Install Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: ${{ env.JAVA_VERSION }}
```

```suggestion
          java-version: ${{ matrix.version }}
```

You missed the whole point of using the job matrix ^^